### PR TITLE
TEST: remove COV_* from environment before testing Popen

### PR DESCRIPTION
**Pull request by mattip**
travis started failing a non-related test in PR #12842. I could reproduce the failure locally when running tests with `python runtests.py --coverage --mode full -t numpy/distutils/tests/test_exec_command.py`. Removing the `COV_CORE_SOURCE`, `COV_CORE_CONFIG`, `COV_CORE_DATAFILE` environment variables before running `Popen([sys.executable ...` fixes it. Perhaps it is due to the `include` line in [`.coveragerc`](https://github.com/numpy/numpy/blob/v1.16.0/.coveragerc#L3), but it seemed safer to treat the failure where it happened.

I don't understand why the failure began happening now. Perhaps one of the coverage-related packages was updated in apt or on PyPI

### CODE REVIEW

**Review by tylerjereddy — COMMENTED** [review]
As far as I can tell all the test codepaths run through `exec_command`, which we've now completely removed from usage in our library.

Do we need to deprecate `exec_command` before we stop testing it, even though we no longer use it in our code base? There's an unresolved discussion of this matter in #11980, which I will bump again now. There's such an obvious standard library replacement for this now.

In short, this fix is probably fine & even if it is not perfect we don't use the function anymore and we're almost certainly going to discourage others from using it, and this doesn't actually change the function itself so if it helps that's fine by me.

That said, I'm not consistently seeing the failure on recent PRs or even on the most recent version of the PR you've linked.
